theano/gpuarray/subtensor.py

```diff
@@ -405,15 +405,15 @@ class GpuAdvancedSubtensor1(HideC, tensor.AdvancedSubtensor1):
     """
     def make_node(self, x, ilist):
         ctx_name = infer_context_name(x, ilist)
-        x_ = as_gpuarray_variable(x, ctx_name)
+        x_ = gpu_contiguous(as_gpuarray_variable(x, ctx_name))
```

Comment:
This is too strict. If the outer dimensions isn't strided, it should still work.

This also need tests.